kratos/spatial_containers/search_wrapper.h

```diff
@@ -0,0 +1,1024 @@
+
+    /**
+     * @brief This method checks if a point is inside any bounding box of the global bounding boxes considering a certain tolerance
+     * @param rCoords The coordinates of the point
+     * @param Tolerance The tolerance
+     * @return The index of the ranks inside the bounding box
+     */
+    std::vector<int> RansksPointIsInsideBoundingBoxWithTolerance(
+        const array_1d<double, 3>& rCoords,
+        const double Tolerance
+        );
+
+    /**
+     * @brief This method takes a point and finds all of the objects in the given radius to it (iterative version) (serial version).
+     * @details The result contains the object and also its distance to the point.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param Radius The radius to be checked
+     * @param rResults The results of the search
+     * @param ClearSolution Clear the current solution
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<class TPointIteratorType>
+    void SerialSearchInRadius(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        const double Radius,
+        ResultContainerVectorType& rResults,
+        const bool ClearSolution = true
+        )
+    {
+        // Clear current solution
+        if (ClearSolution) {
+            rResults.Clear();
+        }
+
+        // Retrieving parameters
+        const int allocation_size = mSettings["allocation_size"].GetInt();
+
+        // Retrieving the number of points
+        const std::size_t number_of_points = itPointEnd - itPointBegin;
+
+        // Definition of indexes
+        std::vector<IndexType> indexes(number_of_points, 0);
+        IndexPartition<IndexType>(number_of_points).for_each([&indexes, &itPointBegin](std::size_t index) {
+            auto it_point = itPointBegin + index;
+            if constexpr (std::is_same<TPointIteratorType, ModelPart::NodeIterator>::value || std::is_same<TPointIteratorType, ModelPart::NodeConstantIterator>::value) {
+                indexes[index] = it_point->Id();
+            } else {
+                indexes[index] = index;
+            }
+        });
+
+        // Initialize results
+        rResults.InitializeResults(indexes);
+
+        // Adding the results to the container
+        IndexPartition<IndexType>(number_of_points).for_each([this, &indexes, &itPointBegin, &rResults, &Radius, &allocation_size](std::size_t index) {
+            auto it_point = itPointBegin + index;
+            auto& r_point_result = rResults[indexes[index]];
+
+            // Search
+            std::vector<ResultType> results;
+            LocalSearchInRadius(*it_point, Radius, results, allocation_size);
+            for (auto& r_result : results) {
+                r_point_result.AddResult(r_result);
+            }
+        });
+
+        // Synchronize
+        rResults.SynchronizeAll(mrDataCommunicator);
```

Comment:
if its 'serial' remove this, otherwise change the name to `LocalSearchInRadius`

Reply:
I cannot as the containervrequires it even in serial

Reply:
a ok I understand now, nevermind.